vcf.c

```diff
@@ -3295,7 +3295,8 @@ static int vcf_idx_init(htsFile *fp, bcf_hdr_t *h, int min_shift, const char *fn
         fp->idx = NULL;
         return -1;
     }
-    fp->fnidx = fnidx;
+    fp->fnidx = strdup(fnidx);
+    if (!fp->fnidx) return -1;
```

Comment:
Should tear down `fp->idx` like the other error path does. Ditto in `bcf_idx_init` below.

Reply:
The top caller (`test_view::main`), does the clean-up of the `htsFILE` descriptor and the memory associated with its members, so calling `hts_idx_destroy` here was redundant. Also, there is no other place where `fp->idx` is subsequently used.

Reply:
No.

If `fp->idx` is not NULL, then the index gets built up on the fly in `bcf_write`/`vcf_write`. When `bcf_idx_init()` fails for any reason, it needs to tear down the `fp->idx` that it has attempted to construct.

Reply:
Well ideally the caller would inspect the return value of `bcf_idx_init()` and not carry on after it failed.  But as we know some callers aren't very good at doing that, it would be best to tidy everything up here.  It could be made slightly easier by moving the `strdup()` to before `hts_idx_init()`.  Both failure paths after that would need to free `fp->fnidx` and set it to `NULL`, but otherwise remain as they were.